src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspace.cs

```diff
@@ -34,6 +41,22 @@ internal VisualStudioWorkspace(HostServices hostServices, WorkspaceBackgroundWor
             }
         }
 
+        private void OnCacheFlushRequested(object sender, EventArgs e)
+        {
+            if (_backgroundCompiler != null)
+            {
+                _backgroundCompiler.Dispose();
+                _backgroundCompiler = null; // PartialSemanticsEnabled will now return false
```

Comment:
It seems that the only place that still uses FrozenPartialSemantics is data tips.  This might mean that calculating datatips get significantly more expensive, as it will now block while metadata is read, files are parsed, skeleton assemblies generated, etc, instead of returning info based on whatever the background compiler has already completed.